libpromises/evalfunction.c

```diff
@@ -2202,7 +2212,7 @@ static FnCallResult FnCallGetIndicesClassic(EvalContext *ctx, ARG_UNUSED const P
 
 static FnCallResult FnCallGetIndices(EvalContext *ctx, ARG_UNUSED const Policy *policy, const FnCall *fp, const Rlist *finalargs)
 {
-    const char *name_str = RlistScalarValue(finalargs);
+    const char *name_str = RlistScalarValueSafe(finalargs);
```

Comment:
`RlistScalarValueSafe()` seems like it should be used in cases where we will print the resulting string (`[not printable]`), but here it's used for actual variable lookup, which seems wrong to me. A more correct approach would be to not perform the variable lookup at all if we don't expect an actual variable, don't you agree?